R/utils.R

```diff
@@ -1341,10 +1332,13 @@ interpolate_str <- function(index_template, ...) {
   return(template)
 }
 
-validate_keys <- function(string) {
+validate_keys <- function(string, is_template) {
```

Comment:
Does the signature change impact existing calls to `validate_keys` that only have a single argument?

Reply:
@josegonzalez `validate_keys` is an internal function, and only called from two Dash methods:
- `interpolate_index`
- `index_string`

The change makes it possible to use the same function for both of those methods, which pass different arguments to `validate_keys` (one passes the page index as a string with interpolation keys included, e.g. `{%app_entry%}` while the other passes in a vector of strings, in which the names of the keys are included, e.g. `app_entry`.

Both are matched against a list of necessary keys, and a useful error message thrown if the required keys are not present.